web/src/upload.js

```diff
@@ -450,15 +478,48 @@ export function initialize() {
         }
     });
 
+    let drag_counter = 0;
     // Allow the app panel to receive drag/drop events.
-    $(".app, #navbar-fixed-container").on("dragover", (event) => event.preventDefault());
+    $(".app, #navbar-fixed-container").on("dragover", (event) => {
+        event.preventDefault();
+        // Show the upload overlay during drag events.
+        $(".upload-overlay-container").addClass("upload-overlay-visibility");
+        // Adjust z-index so that message box is visible.
+        if ($(".message_comp").css("display") === "block") {
```

Comment:
It's regrettable the compose box doesn't have a state class for its visibility. That's outside the scope of this PR, but it's something that ought to be addressed as part of the compose-box redesign project.